cpp/include/cudf/column/dlpack.hpp

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cudf/column/column.hpp>
+
+#include <dlpack/dlpack.h>
+
+namespace cudf {
+
+/**
+ * @brief Convert a DLPack DLTensor into cudf::columns
+ *
+ * @note The managed tensor is not deleted by this function.
+ *
+ * @throw TODO
+ * 
+ * @param managed_tensor a 1D or 2D column-major (Fortran order) tensor
+ * @param mr Optional resource to use for device memory allocation
+ * 
+ * @return Columns converted from DLPack
+ */
+std::vector<std::unique_ptr<column>> from_dlpack(
```

Comment:
Should this return a table?

Reply:
Also possible that we could return a table view of column_views that index into the original DLManagedTensor memory rather than copying it?

Reply:
Yeah, this should just be a table, I'll fix that.

Reply:
Yes, I could also just return views into the DLManagedTensor. As the managed tensor would still own the memory, it would need to be kept alive while the view is being used. Also, would I need two versions, one each for table_view and mutable_table_view?

Reply:
Hmmm. I don't know the answers. This feels like something that could be a future optimization. I would say just get the port working with a table for now.